src/System.Private.CoreLib/shared/System/IO/BinaryReader.cs

```diff
@@ -649,6 +578,46 @@ public virtual byte[] ReadBytes(int count)
             return result;
         }
 
+        private ReadOnlySpan<byte> InternalRead(int numBytes)
+        {
+            Debug.Assert(numBytes >= 2 && numBytes <= 16, "value of 1 should use ReadByte. value > 16 requires to change the minimal _buffer size");
+
+            if (_isMemoryStream)
+            {
+                // no need to check if _stream == null as we will never have null _stream when _isMemoryStream = true
+
+                // read directly from MemoryStream buffer
+                Debug.Assert(_stream is MemoryStream);
+                return ((MemoryStream)_stream).InternalReadSpan(numBytes);
+            }
+            else
+            {
+                if (_stream == null)
+                {
+                    throw Error.GetFileNotOpen();
+                }
+
+                int bytesRead = 0;
+                int n = 0;
```

Comment:
FillBuffer used ReadByte when numBytes == 1.  Is there a reason that path wasn't brought forward to InternalRead? (Sorry if this was discussed previously and I missed it.)

Reply:
Oh, I see, it's now never called with numBytes == 1.